experimental/unitsconversion/src/info_provider.rs

```diff
@@ -0,0 +1,160 @@
+    }
+    icu_unitsconversion_data::make_provider!(Baked);
+    icu_unitsconversion_data::impl_units_info_v1!(Baked);
+};
+
+#[cfg(feature = "datagen")]
+/// The latest minimum set of keys required by this component.
+pub const KEYS: &[DataKey] = &[UnitsInfoV1Marker::KEY];
+
+/// This type encapsulates all the constant data required for unit conversions.
+///
+/// <div class="stab unstable">
+/// 🚧 This code is considered unstable; it may change at any time, in breaking or non-breaking ways,
+/// including in SemVer minor releases. While the serde representation of data structs is guaranteed
+/// to be stable, their Rust representation might not be. Use with caution.
+/// </div>
+#[icu_provider::data_struct(marker(UnitsInfoV1Marker, "units/info@1", singleton))]
+#[derive(Clone, PartialEq, Debug)]
+#[cfg_attr(
+    feature = "datagen",
+    derive(serde::Serialize, databake::Bake),
+    databake(path = icu_unitsconversion::units_provider),
+)]
+#[cfg_attr(feature = "serde", derive(serde::Deserialize))]
+#[yoke(prove_covariance_manually)]
+pub struct UnitsInfoV1<'data> {
+    /// Maps from unit name (e.g. foot) to the index of the unit in the `unit_quantity` vector.
+    #[cfg_attr(feature = "serde", serde(borrow))]
+    pub units_info: ZeroMap<'data, str, UnitsInfoIndex>,
+
+    /// Contains the quantity information for the units.
+    /// For example, the quantity for the unit `foot` is `length`.
+    #[cfg_attr(feature = "serde", serde(borrow))]
+    pub unit_quantity: VarZeroVec<'data, UnitQuantityULE>,
+
+    /// Contains the conversion information, such as the conversion rate and the base unit.
+    /// For example, the conversion information for the unit `foot` is `1 foot = 0.3048 meter`.
+    #[cfg_attr(feature = "serde", serde(borrow))]
+    pub convert_units: VarZeroVec<'data, ConvertUnitsULE>,
+}
+
+#[zerovec::make_ule(UnitsInfoIndexULE)]
+#[cfg_attr(
+    feature = "datagen",
+    derive(databake::Bake),
+    databake(path = icu_unitsconversion::units_provider),
+)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+#[derive(Copy, Debug, Clone, Default, PartialEq, PartialOrd, Eq, Ord)]
+pub struct UnitsInfoIndex {
+    /// Contains the index of the quantity in the `unit_quantity` vector.
+    /// If the unit does not have a quantity, this field is `None`.
+    pub quantity: Option<u16>,
+
+    /// Contains the index of the convert unit in the `convert_units` vector.
+    /// If the unit does not have a convert unit, this field is `None`.
+    pub convert_unit: Option<u16>,
+}
+
+#[zerovec::make_ule(QuantitySimplicityULE)]
+#[cfg_attr(
+    feature = "datagen",
+    derive(serde::Serialize, databake::Bake),
+    databake(path = icu_unitsconversion::units_provider),
+)]
+#[cfg_attr(feature = "serde", derive(serde::Deserialize))]
+#[derive(Copy, Clone, Debug, PartialOrd, Ord, PartialEq, Eq, Default)]
+#[repr(u8)]
+pub enum QuantitySimplicity {
+    #[default]
+    Simple = 0,
+    Complex = 1,
```

Comment:
doc? I have no idea what this could mean

Reply:
done